v03_pipeline/lib/reference_datasets/reference_dataset.py

```diff
@@ -0,0 +1,93 @@
+from enum import Enum
+from types import MappingProxyType
+
+from v03_pipeline.lib.model import AccessControl, DatasetType, Env, ReferenceGenome
+from v03_pipeline.lib.reference_datasets.cadd import load_cadd_ht_from_raw_dataset
+from v03_pipeline.lib.reference_datasets.hgmd import (
+    HGMD_ENUM_SELECT,
+    download_and_import_hgmd_vcf,
+)
+
+LOAD_DATASET_FUNC = 'load_parsed_dataset_func'
+VERSION = 'version'
+RAW_DATASET_PATH = 'raw_dataset_path'
+ENUM_SELECT = 'enum_select'
+
+
+class ReferenceDataset(str, Enum):
+    cadd = 'cadd'
+    hgmd = 'hgmd'
+
+    CONFIG = MappingProxyType(
+        {
+            'cadd': {
+                LOAD_DATASET_FUNC: load_cadd_ht_from_raw_dataset,
+                ReferenceGenome.GRCh37: {
+                    VERSION: '1.0',
+                    RAW_DATASET_PATH: [
+                        'https://krishna.gs.washington.edu/download/CADD/v1.7/GRCh37/whole_genome_SNVs.tsv.gz',
+                        'https://krishna.gs.washington.edu/download/CADD/v1.7/GRCh38/gnomad.genomes.r4.0.indel.tsv.gz',
+                    ],
+                },
+                ReferenceGenome.GRCh38: {
+                    VERSION: '1.0',
+                    RAW_DATASET_PATH: [
+                        'https://krishna.gs.washington.edu/download/CADD/v1.7/GRCh38/whole_genome_SNVs.tsv.gz',
+                        'https://krishna.gs.washington.edu/download/CADD/v1.7/GRCh37/gnomad.genomes-exomes.r4.0.indel.tsv.gz',
+                    ],
+                },
+            },
+            'hgmd': {
+                LOAD_DATASET_FUNC: download_and_import_hgmd_vcf,
+                ReferenceGenome.GRCh37: {
+                    VERSION: '1.0',
+                    RAW_DATASET_PATH: 'gs://seqr-reference-data/v3.1/GRCh37/reference_datasets/hgmd-v2020.1.vcf.bgz',
+                    ENUM_SELECT: HGMD_ENUM_SELECT,
+                },
+                ReferenceGenome.GRCh38: {
+                    VERSION: '1.0',
+                    RAW_DATASET_PATH: 'gs://seqr-reference-data/v3.1/GRCh38/reference_datasets/hgmd-v2020.1.vcf.bgz',
+                    ENUM_SELECT: HGMD_ENUM_SELECT,
```

Comment:
is this actually used anywhere? If not, it might be worth pulling HGMD out of this PR to limit the scope and then add it back in its wown PR that actually implements the enum select functionality

Reply:
also, looking at the current config `enum_select` is identical across reference genomes every place it is used so I doo not think we should be nesting it under the build

Reply:
noted, and I'll leave hgmd out